ios/FluentUI/Core/Colors2.swift

```diff
@@ -0,0 +1,181 @@
+        brandColors[.brandStroke1] = brandStroke1
+    }
+    if let brandStroke1Pressed = provider.brandStroke1Pressed(for: theme)?.dynamicColor {
+        brandColors[.brandStroke1Pressed] = brandStroke1Pressed
+    }
+    if let brandStroke1Selected = provider.brandStroke1Selected(for: theme)?.dynamicColor {
+        brandColors[.brandStroke1Selected] = brandStroke1Selected
+    }
+    return brandColors
+}
+
+// MARK: Colors
+
+public enum BrandColorsForOverriding: CaseIterable {
+    case brandBackground1
+    case brandBackground1Pressed
+    case brandBackground1Selected
+    case brandBackground2
+    case brandBackground2Pressed
+    case brandBackground2Selected
+    case brandBackground3
+    case brandBackgroundTint
+    case brandBackgroundDisabled
+    case brandForeground1
+    case brandForeground1Pressed
+    case brandForeground1Selected
+    case brandForegroundTint
+    case brandForegroundDisabled1
+    case brandForegroundDisabled2
+    case brandStroke1
+    case brandStroke1Pressed
+    case brandStroke1Selected
+
+    var equivalentAliasToken: AliasTokens.ColorsTokens {
+        switch self {
+        case .brandBackground1:
+            return .brandBackground1
+        case .brandBackground1Pressed:
+            return .brandBackground1Pressed
+        case .brandBackground1Selected:
+            return .brandBackground1Selected
+        case .brandBackground2:
+            return .brandBackground2
+        case .brandBackground2Pressed:
+            return .brandBackground2Pressed
+        case .brandBackground2Selected:
+            return .brandBackground2Selected
+        case .brandBackground3:
+            return .brandBackground3
+        case .brandBackgroundTint:
+            return .brandBackgroundTint
+        case .brandBackgroundDisabled:
+            return .brandBackgroundDisabled
+        case .brandForeground1:
+            return .brandForeground1
+        case .brandForeground1Pressed:
+            return .brandForeground1Pressed
+        case .brandForeground1Selected:
+            return .brandForeground1Selected
+        case .brandForegroundTint:
+            return .brandForegroundTint
+        case .brandForegroundDisabled1:
+            return .brandForegroundDisabled1
+        case .brandForegroundDisabled2:
+            return .brandForegroundDisabled2
+        case .brandStroke1:
+            return .brandStroke1
+        case .brandStroke1Pressed:
+            return .brandStroke1Pressed
+        case .brandStroke1Selected:
+            return .brandStroke1Selected
+        }
+    }
+}
+
+public extension FluentTheme {
+    /// Associates a `ColorProvider2` with a given `FluentTheme` instance.
+    ///
+    /// - Parameters:
+    ///   - provider: The `ColorProvider2` whose colors should be used for controls in this theme.
+    ///   - theme: The theme where these colors should be applied.
+    @objc static func setProvider(provider: ColorProviding2, for theme: FluentTheme) {
```

Comment:
Since this is an extension to `FluentTheme`, could we remove the theme parameter and call self inside the function instead?

Reply:
Good call - that makes this simpler to call, as well.